core/src/main/java/com/cloud/agent/resource/virtualnetwork/VirtualRoutingResource.java

```diff
@@ -173,12 +174,33 @@ private Answer execute(final SetupKeyStoreCommand cmd) {
         return new SetupKeystoreAnswer(result.getDetails());
     }
 
+    private void scpFileToIdentifyPatching(String routerIp, String path, String filename, String content) throws InterruptedException {
+        String errMsg = "Failed to scp file: %s to system VM";
+        for (int retries = 15; retries > 0; retries--) {
+            try {
+                _vrDeployer.createFileInVR(routerIp, path, filename, content);
+            } catch (Exception e) {
+                errMsg += ", retrying";
+                s_logger.error(String.format(errMsg, filename), e);
+                Thread.sleep(2000);
+            }
+        }
+    }
     private Answer execute(final SetupCertificateCommand cmd) {
-        final String args = String.format("/usr/local/cloud/systemvm/conf/agent.properties " +
+        String routerName = cmd.getAccessDetail(NetworkElementCommand.ROUTER_NAME);
+        if (!org.apache.commons.lang3.StringUtils.isEmpty(routerName) && (routerName.startsWith("s-") || routerName.startsWith("v-"))) {
+            try {
+                scpFileToIdentifyPatching(cmd.getRouterAccessIp(), "/usr/local/cloud/systemvm/conf/", KeyStoreUtils.CERT_FILENAME, cmd.getCertificate());
```

Comment:
tangent idea - I was thinking would it be even simpler to create the `KeyStore` in memory by the mgmt server, convert that to bytes/base64-encoded data that is then serialised/saved directly in the systemvm without the hassel of import scripts in systemvm?

Reply:
Sorry @rohityadavcloud I didn't quite understand this. Do you mean pass it to the systemvm via the cmdline args?

Reply:
@rohityadavcloud could you please explain this point. Thanks